components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/util/JdbcUtils.java

```diff
@@ -17,74 +17,78 @@
 package org.wso2.carbon.identity.application.mgt.util;
 import org.wso2.carbon.database.utils.jdbc.JdbcTemplate;
 import org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException;
-import org.wso2.carbon.identity.core.persistence.JDBCPersistenceManager;
-
-import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.DB2;
-import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.H2;
-import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.INFORMIX;
-import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.MARIADB;
-import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.MICROSOFT;
-import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.MY_SQL;
-import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.POSTGRE_SQL;
-import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.S_MICROSOFT;
 
 /**
  * A util class to support the Jdbc executions.
+ * @deprecated use {@link org.wso2.carbon.identity.core.util.JdbcUtils} instead.
  */
 public class JdbcUtils {
 
     /**
      * Get a new Jdbc Template.
      *
      * @return a new Jdbc Template.
+     * @deprecated use {@link org.wso2.carbon.identity.core.util.JdbcUtils#getNewTemplate()} instead.
      */
+    @Deprecated
     public static JdbcTemplate getNewTemplate() {
 
-        return new JdbcTemplate(JDBCPersistenceManager.getInstance().getDataSource());
+        return org.wso2.carbon.identity.core.util.JdbcUtils.getNewTemplate();
     }
 
     /**
      * Check if the DB is H2, MySQL or Postgres.
      *
      * @return true if DB is H2, MySQL or Postgres, false otherwise.
      * @throws DataAccessException if error occurred while checking the DB metadata.
+     * @deprecated
      */
+    @Deprecated
     public static boolean isH2MySqlOrPostgresDB() throws DataAccessException {
 
-        return isDBTypeOf(MY_SQL) || isDBTypeOf(H2) || isDBTypeOf(POSTGRE_SQL) || isDBTypeOf(MARIADB);
+        return org.wso2.carbon.identity.core.util.JdbcUtils.isMySQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isH2DB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isPostgreSQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isMariaDB();
```

Comment:
shall we add static imports for these? Would improve readability

Reply:
Addressed by https://github.com/wso2/carbon-identity-framework/pull/3993